From a620eb30751fa921f1d8d00f0511e267e998859a Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Wed, 15 Sep 2010 15:47:20 +0300
Subject: [PATCH] Workaround tab label related crashes.

Try to workaround tab label related crashes with recent xulrunners.
Its not a fix but a bandaid.
---
 src/galeon-tab-menu.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/galeon-tab-menu.c b/src/galeon-tab-menu.c
index 05d9eaa..f6640d2 100644
--- a/src/galeon-tab-menu.c
+++ b/src/galeon-tab-menu.c
@@ -71,6 +71,8 @@ update_title_cb (GaleonEmbed* embed, GtkAction *action)
 {
 	GaleonTab *tab = g_object_get_data (G_OBJECT (action), "GaleonTab");
 
+	g_return_if_fail (GALEON_IS_TAB (tab));
+
 	const char *title = galeon_tab_get_title(tab);
 	gchar *title_s = gul_string_shorten(title,
 					    MENU_ITEM_MAX_LENGTH);
-- 
1.7.2.3

